### ISSUE TITLE
Ligatures, Org-mode hide leading stars, and Emacs 28

### ISSUE THREAD

**Issue by jcf** [hlissner/doom-emacs#4922]
**TL;DR** I'm looking into how to remove the triple-asterisk ligature from Org-mode with Emacs 28 because I like leading "stars" to be hidden. I've found the `composition-function-table`, and the module docs on adding ligatures, just not sure if this is the desired behaviour… for now I'm working around the issue by enabling `+pretty`, with the following org-superstar configuration:

``` elisp
(after! org-superstar
  (setq
   org-superstar-headline-bullets-list '("#")
   org-superstar-special-todo-items nil))
```

**What did you expect to happen?**

With `org-hide-leading-stars` set to true, I'd expect leading stars to only show up as a single asterisk.

**What actually happened?**

Multiples of three asterisks appear as a cute ligature that kinda looks like a crown (with [JetBrains Mono](https://www.jetbrains.com/lp/mono/)).

<img width="133" alt="image" src="https://user-images.githubusercontent.com/18374/115988312-fea53880-a5b0-11eb-9fa6-6e0ebb60f6ee.png">

**Additional details:**

This happens with both `ligatures` and `(ligatures +extra)` enabled, on both macOS and Linux, built from a recent native-comp Emacs master.

```
GNU Emacs 28.0.50 (build 1, x86_64-apple-darwin19.6.0, NS appkit-1894.60 Version 10.15.7 (Build 19H114)) of 2021-01-06
```

If I add `+pretty` to my Org module flags, the pretty stars plugin replaces the plain asterisks, and the ligatures are no longer applied.

```
(org +dragndrop +gnuplot +noter +pandoc +pomodoro +present +roam)
```

**System information:**

https://pastebin.com/ZmFY8jeL

**Comment by gagbo** [comment]
You basically won't have anyway of changing this without getting your feet wet, and I (very personal opinion) shouldn't be changed in doom-emacs but left to the tinkerer.

Changing the value of the composition function table alist
https://github.com/hlissner/doom-emacs/blob/f621ff80471e8d08a72e5ece00641c70b121873a/modules/ui/ligatures/config.el#L50-L79
to something that only matches the ligatures that you want, in the modes that you want is going to be the only way to exclude ligatures included in your font from being considered for ligation by Emacs. The "ugly" regexes have comments with the human readable comment that generated them, so you can look for the entry that has `?*` as key, redo the `regex-opt` call with the ligature excluded, and then replace the value in the variable.

**Comment by jcf** [comment]
This is really helpful. Thanks, @gagbo!

Out of interest, why do you think the `*` behaviour in Doom is right and should be undone by the tinkerer rather than the other way around? Shouldn't the tinkerer add the regex that comes out of the box if it interacts poorly with a relatively widely used combination of font and mode?

**Comment by gagbo** [comment]
> Out of interest, why do you think the `*` behaviour in Doom is right and should be undone by the tinkerer rather than the other way around? Shouldn't the tinkerer add the regex that comes out of the box if it interacts poorly with a relatively widely used combination of font and mode?

Mostly because the `*` automatic ligatures is very often something you want (i.e. _not_ in org-mode). For the users that don't use org-mode, or even for people who only use org-mode a little, it makes little sense to deactivate it, whereas extensive org-mode users can change that behaviour. So my point of view is mostly "all org-mode users are Emacs users, but not all Emacs users are org-mode users".

There are workarounds for that (programmatically make the ligatures not link `*` in org-mode buffers), and it's what I started in another PR, but it's a little harder to finish
